### ISSUE TITLE
WINNF.FT.D.DRG.2 requires relinquishment after deregistration - not  according to SAS spec

### ISSUE THREAD

**Issue by meitalk** [Wireless-Innovation-Forum/Citizens-Broadband-Radio-Service-Device#122]
Test WINNF.FT.D.DRG.2 requires to trigger deregisteration request for granted and transmitting cbsds.   Then it expected from the DP to send relinquishment and deregistration request.
According top SAS spec when sending deregistration request, the DP is not required to send relinquishment request – in case of deregistration the CBSD grant will automatically be relinquished.

"When the CBSD determines that it should deregister from the SAS, it shall cease transmission associated with any Grants and then shall send a DeregistrationRequest object (cbsdId) to the SAS. The CBSD then cancels any Grants that it still believes are allocated to it. 
The SAS marks the CBSD as Unregistered, removes any existing Grants, and responds with a DeregistrationResponse object (cbsdId, response)."
==================================
1.1.1.1.1	[WINNF.FT.D.DRG.2] 

1. Ensure the following conditions are met for test entry:
•	Each UUT has successfully registered with SAS Test Harness
•	Each UUT is in the authorized state
Invoke trigger to deregister each UUT from the SAS, UUT shall cease transmission associated with any Grants
2. UUT sends a valid Relinquishment request and receives Relinquishment response with responseCode=0
3. DP with two CBSDs sends Deregistration Request with two DeregistrationRequest object to the SAS Test Harness.
This may occur in a separate message per CBSD, or together in a single message with array of 2.
The two objects should have cbsdIds C1 and C2 in that order.

**Comment by dgcell20** [comment]
WINNF-TS-0016 v1.2.1 further states in section 8.8.1 that "The CBSD should send a RelinquishmentRequest object for each Grant prior to sending the DeregistrationRequest object."
Though this uses "should" and not "shall", so I agree with relinquishment is not required.
The WInnForum SAS Test Harness allows for the relinquishment messages to be optional, refer to section 13.7 of the WINNF-IN-0156 WInnForum SAS Test Harness CBSD UUT Tutorial v1.0.0.1.

Maybe WINNF-TS-0122 could have some better wording, but note that step 2 (UUT sends a Relinquishment request and receives Relinquishment response) does not have a Pass/Fail criteria.

Are you commenting just about the wording, or is the test failing because the Relinquishment request/response does not exist?

**Comment by cwilliams-ericsson** [comment]
I agree with your assessment -- the DRG test cases in WINNF-TS-0122-V1.0.0 are not completely clear.  However, the test harness supports RLQ as an optional message in test cases (UUT will pass whether it sends a RLQ or not before the DRG).

I think this is not a test harness issue, but is a test document wording issue.  Therefore, I suggest that this   should be brought up at the regular  WINNF WG4 CBSD meeting, and not as a GitHub issue.

**Comment by meitalk** [comment]
The test passes without relinquishment. The wording in the section 2 of test WINNF.FT.D.DRG.2 was not clear. It was not clear if step 2 is mandatory or optional.

**Comment by awaizkhan** [comment]
proposed resolution to make a change in 0122 doc for all DRG test cases
UUT "may" sends a Relinquishment request and receives Relinquishment
response with responseCode=0

**Comment by dgcell20** [comment]
Wording has been changed in WINNF-TS-0122-V1.0.1 (test cases WINNF.FT.C.DRG.1, D.DRG.2, C.DRG.3, D.DRG.4)
https://workspace.winnforum.org/higherlogic/ws/public/document?document_id=7341
